File: simple-record-ws/Simple-Record.Application/Services/Contracts/IPersonServices.cs

using simple_record.service.InputModels;
using simple_record.service.Models;
using simple_record.service.ViewModels;

namespace simple_record.service.Services.Contracts
{
    public interface IPersonServices
    {
        Task<GenericServiceResult> CreatePerson(CreatePersonInputModel model);
        Task<GenericServiceResult> GetAllPeoples(GetAllPeoplesInputModel model);
        Task<GenericServiceResult> GetPersonById(GetPersonByIdInputModel model);
        Task<GenericServiceResult> UpdatePerson(UpdatePersonInputModel model);
        Task<GenericServiceResult> DeletePerson(DeletePersonInputModel model);
    }
}

File: simple-record-app/src/core/Contracts/IPersonServices.ts
import { CreatePersonInputModel } from "../InputModels/CreatePersonInputModel";
import { DeletePersonInputModel } from "../InputModels/DeletePersonInputModel";
import { GetAllPeoplesInputModel } from "../InputModels/GetAllPeoplesInputModel";
import { GetPersonByIdInputModel } from "../InputModels/GetPersonByIdInputModel";
import { UpdatePersonInputModel } from "../InputModels/UpdatePersonInputModel";
import { GenericServiceResult } from "../ViewModels/GenericServiceResult";

export interface IPersonServices {
    CreatePerson(data: CreatePersonInputModel): Promise<GenericServiceResult>
    GetAllPeoples(data: GetAllPeoplesInputModel): Promise<GenericServiceResult>
    GetPersonById(data: GetPersonByIdInputModel): Promise<GenericServiceResult>
    UpdatePerson(data: UpdatePersonInputModel): Promise<GenericServiceResult>
    DeletePerson(data: DeletePersonInputModel): Promise<GenericServiceResult>
}